andela/ah-backend-realers
# 166187854 send verification email upon successful registraion

#### What does this PR do?
- send verification email upon successful registration

#### Description of Task to be completed?

- add email configuration
- add a method to generate verification link
- add a method to send a verification link
- add functionality to verify and activate a user
- add tests for the send verification functions
- improve test coverage
- add the staticfiles folder to fix staticfiles warning

#### How should this be manually tested?
- fetch this repo locally in terminal with `$ git fetch origin ft-verifcation-email-166187854`
- do `$ pip install -r requirements.txt` to install the added package dependency
- add the lines below to your env variable files and set them according the slack message I shared.
```
export SECURITY_PASSWORD_SALT='any_security_password_salt'
export EMAIL_HOST='your_email_host_used'
export EMAIL_USE_TLS='boolean (True/False)'
export EMAIL_PORT='your_email_port
export EMAIL_HOST_USER='you_email_address'
export HOST_PASSWORD='your_email_password'
```
- Now, run the application and then test `registration api`. upon successful registration, you will be asked to check your email for verification link as you may see below
<img width="639" alt="Screenshot 2019-06-08 at 23 50 15" src="https://user-images.githubusercontent.com/21230051/59152210-53387c80-8a48-11e9-9508-87d55326cac0.png">

- The you should be able to see the email with the link.
<img width="1097" alt="Screenshot 2019-06-08 at 23 52 08" src="https://user-images.githubusercontent.com/21230051/59152221-8975fc00-8a48-11e9-97a9-055d1b45a7ef.png">

- After clicking on the link, you receive a message to confirm to you that the account has been activated.
<img width="1168" alt="Screenshot 2019-06-08 at 23 53 48" src="https://user-images.githubusercontent.com/21230051/59152243-ccd06a80-8a48-11e9-8da9-4b5a82c55a68.png">

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?

- Story ID - [166187854](https://www.pivotaltracker.com/story/show/166187854)


#### DEV NOTE
- To have the tests to run ensure to edit your **SECRET_KEY** & **SECURITY_PASSWORD_SALT** to the one I will share in slack
- **Kindly remember to test all of the current APIs both locally and using the review app [link](https://ah-backend-realers-stagi-pr-14.herokuapp.com/)**
